debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java

```diff
@@ -229,25 +230,34 @@
 
     protected List<SchemaAndValueField> schemasAndValuesForArrayTypes() {
        return Arrays.asList(new SchemaAndValueField("int_array", SchemaBuilder.array(Schema.OPTIONAL_INT32_SCHEMA).optional().build(),
-                                Arrays.asList(1, 2, 3)),
+                                expectedArray(1, 2, 3)),
                             new SchemaAndValueField("bigint_array", SchemaBuilder.array(Schema.OPTIONAL_INT64_SCHEMA).optional().build(),
-                                Arrays.asList(1550166368505037572L)),
+                                expectedArray(1550166368505037572L)),
                             new SchemaAndValueField("text_array", SchemaBuilder.array(Schema.OPTIONAL_STRING_SCHEMA).optional().build(),
-                                Arrays.asList("one", "two", "three")),
+                                expectedArray("one", "two", "three")),
                             new SchemaAndValueField("char_array", SchemaBuilder.array(Schema.OPTIONAL_STRING_SCHEMA).optional().build(),
-                                    Arrays.asList("cone      ", "ctwo      ", "cthree    ")),
+                                expectedArray("cone      ", "ctwo      ", "cthree    ")),
                             new SchemaAndValueField("varchar_array", SchemaBuilder.array(Schema.OPTIONAL_STRING_SCHEMA).optional().build(),
-                                    Arrays.asList("vcone", "vctwo", "vcthree"))
+                                expectedArray("vcone", "vctwo", "vcthree")),
+                            new SchemaAndValueField("date_array", SchemaBuilder.array(Date.builder().optional().schema()).optional().build(),
+                                expectedArray(17109, 17110, 17111))
                             );
     }
 
+    private List<?> expectedArray(Object... elements) {
```

Comment:
This is the reason - https://github.com/jpechane/debezium/blob/d01125f43941e253e166b3b975cdeaaa01248c8b/debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java#L400

With mutable array it workds, with immutable it fails and I'd need to introduce a more complex logic for class comparison.

Reply:
Ah, got it. I've adjusted the assertion so that any `List` implementations are accepted as valid in the list case, we don't need to match the exact one. This allowed me to use `Arrays#asList()` again. I've added this commit, also fixing a few typos.